lib/charms/vault_k8s/v0/vault_autounseal.py

```diff
@@ -208,18 +211,38 @@ def restore(self, snapshot: Dict[str, Any]) -> None:
 class VaultAutounsealProvidesEvents(ops.ObjectEvents):
     """Events raised by the vault-autounseal relation on the provider side."""
 
-    vault_autounseal_initialize = ops.EventSource(VaultAutounsealInitialize)
-    vault_autounseal_destroy = ops.EventSource(VaultAutounsealDestroy)
+    vault_autounseal_requirer_relation_created = ops.EventSource(
+        VaultAutounsealRequirerRelationCreated
+    )
+    vault_autounseal_requirer_relation_broken = ops.EventSource(
+        VaultAutounsealRequirerRelationBroken
+    )
 
 
 class VaultAutounsealRequireEvents(ops.ObjectEvents):
     """Events raised by the vault-autounseal relation on the requirer side."""
 
     vault_autounseal_details_ready = ops.EventSource(VaultAutounsealDetailsReadyEvent)
-    vault_autounseal_provider_removed = ops.EventSource(VaultAutounsealProviderRemoved)
+    vault_autounseal_provider_relation_broken = ops.EventSource(VaultAutounsealProviderRemoved)
+
+
+@dataclass
+class AutounsealDetails:
```

Comment:
I'm not sure calling the class `Autounseal` and `Approle` makes sense. These classes don't have or support any functionality for either of those things, and they're merely a container for the information.

Perhaps they could be renamed `AutounsealConfigurationProperties` and `ApproleLoginDetails` or something, but I think removing the `Details` part makes it more confusing, personally. I was under the impression this is a fairly common pattern for these types of classes, but maybe I'm incorrect.

Reply:
This is true of all dataclasses though isn't it? Should add the `Details` suffix to every dataclass? It may be a common pattern, I really don't know.

Reply:
> This is true of all dataclasses though isn't it? Should add the `Details` suffix to every dataclass?

That's a good point, let me revise my argument a little  -- I think it's important to consider what these classes represent. They don't represent Autounseal or Approle. Just the details surrounding their usage.

In most examples of dataclasses I find things like `Person`, `Rectangle`, `InventoryItem`, etc. Those all make sense because they actually represent those things in the context. But the containers we're discussing do not. They're only used to pass chunks of data.  If I saw an object of type `Autounseal`, I would expect it to represent some manifestation of auto-unseal, not merely the configuration details needed to set it up in the vault config.

Reply:
@gruyaume I don't have a super strong opinion here, but enough to push back a little. If I still haven't convinced you I'll change the names. Let me know.

Reply:
Good with me!